src/tcp_interface.cpp

```diff
@@ -107,8 +100,9 @@ return_statuses TCPInterface::read(unsigned char *msg,
                                  boost::asio::placeholders::error));
   }
 
+  msg->assign(10000, 0);
```

Comment:
Assigning a huge buffer to a vector and then treating it as a buffer is fine, but might be better to use vector::reserve and then vector:shrink_to_fit. That way there's less unneeded memory that stays allocated. Either way you'll have to manage msg.size manually since this call to assign sets msg.size to 10000.

Reply:
Since shrink_to_fit just shrinks the allocated memory to fit the size and reserve doesn't increase the size, this will still result in a vector with size == 0 and 0 memory reserved. Allocation through assign, push_back, or emplace are the only ways to make the vector size actually grow. Since the actual number of bytes read is returned in this function, I figured I would just assign something huge and the user will know how many bytes are valid through a read.

Reply:
You're right. It'd be better to call vector::resize(bytes_read) instead of vector::shrink_to_size(). If we're not going to actually use the dynamic allocation, why are we using vectors?

Reply:
But you can't because it needs the buffer before the bytes have been read. Let me look more into how boost::buffer with a vector works. It might do the allocation itself and this may not be necessary.

Reply:
So, I'm reading over this code more and I think this call to async_read is a terrible idea as implemented. I think I did it because you can only provide a timeout when doing an async read but doing it as implemented means that the vector that gets passed in is managed by the API user and they would have to know that they can't modify the vector until the read returns, which isn't currently being triggered. I need to rethink this and implement it differently. I'm going to cancel this PR until that is done.

Reply:
Used reserve() instead and the boost::buffer allocates the bytes as necessary.